src/Toolbox.php

```diff
@@ -2087,84 +2095,101 @@ public static function createSchema($lang = 'en_GB', ?DBmysql $database = null)
+                }
+                $res = $stmt->execute();
+                if (false === $res) {
+                    $msg = $stmt->error;
+                    $msg .= "\nError execution statement in table $table\n";
+                    $msg .= print_r($row, true);
+                    throw new \RuntimeException($msg);
                 }
             }
+        }
 
-            // Create default forms
-            $default_forms_manager = new DefaultDataManager();
-            $default_forms_manager->initializeData();
+        $progressCallback(null, $number_of_queries, __('Creating default forms…'));
+        $default_forms_manager = new DefaultDataManager();
+        $default_forms_manager->initializeData();
 
-            // Initalize rules
-            RulesManager::initializeRules();
+        $progressCallback(null, $number_of_queries, __('Initalizing rules…'));
+        RulesManager::initializeRules();
 
-            // Make sure keys are generated automatically so OAuth will work when/if they choose to use it
-            \Glpi\OAuth\Server::generateKeys();
+        $progressCallback(null, $number_of_queries, __('Generating keys…'));
+        // Make sure keys are generated automatically so OAuth will work when/if they choose to use it
+        \Glpi\OAuth\Server::generateKeys();
 
-           // update default language
-            Config::setConfigurationValues(
-                'core',
+        $progressCallback(null, $number_of_queries, __('Updating default language…'));
+        Config::setConfigurationValues(
+            'core',
+            [
+                'language'      => $lang,
+                'version'       => GLPI_VERSION,
+                'dbversion'     => GLPI_SCHEMA_VERSION,
+            ]
+        );
+
+        if (defined('GLPI_SYSTEM_CRON')) {
+            $progressCallback(null, $number_of_queries, __('Configuring cron tasks…'));
+           // Downstream packages may provide a good system cron
+            $DB->update(
+                'glpi_crontasks',
                 [
-                    'language'      => $lang,
-                    'version'       => GLPI_VERSION,
-                    'dbversion'     => GLPI_SCHEMA_VERSION,
+                    'mode'   => 2
+                ],
+                [
+                    'name'      => ['!=', 'watcher'],
+                    'allowmode' => ['&', 2]
                 ]
             );
-
-            if (defined('GLPI_SYSTEM_CRON')) {
-               // Downstream packages may provide a good system cron
-                $DB->update(
-                    'glpi_crontasks',
-                    [
-                        'mode'   => 2
-                    ],
-                    [
-                        'name'      => ['!=', 'watcher'],
-                        'allowmode' => ['&', 2]
-                    ]
-                );
-            }
         }
+
+        $progressCallback($number_of_queries, $number_of_queries, __('Done!'));
```

Comment:
I think we should remove the progress messages once the process is done. The 100% progress bar already indicates it.

```suggestion
        $progressCallback($number_of_queries, $number_of_queries, '');
```